### PR TITLE
Fixed updateMean method to work for all inputs

### LINKED ISSUE #40: Fix the updateMean method of MathBasics class to work for all inputs
**Actual Behaviour**

Currently, updateMean method in MathBasics class returns the updated mean for n > 1 and returns val(newly inserted element) as the output for n <= 1. It doesn't throw an error for n <= 0. Here, 'n' refers to number of elements in the array after adding 'val'(newly inserted element).
Also, it has redundant double typecastings in each line.

**Expected Behaviour**

The method should throw an error for n <= 0 as the number of elements in the array cannot be 0. Additionally, all the corresponding calls to this method should catch this exception.
Also, we need to remove extra lines for step wise computation and extra typecasting to double. There is also no need to create additional if branch for n>1.

**Extending the idea**

The whole idea could be extended to other methods in this class too i.e. we can add exception handling for division by zero.
Also, the defined 'n' is number of elements in the array after adding new element which is quite unintuitive. The defined 'n' could be changed to number of elements before adding new element, the same definition that is used in 'updateVariance' method.

**Recommended Changes**

```
public static double updateMean(double currentMean, int n, double val) {
		double nD = (double) n;
		if (n > 1) {
			double result = (double) ((1 / nD) * val);
			result += (double) (((nD - 1) / (nD)) * currentMean);
			return result;
		}

		return val;
}
```
The above could be transformed to below:-

```
public static double updateMean(double currentMean, int n, double val) {
		if(n <= 0)
		    throw new IllegalArgumentException("Number of elements(n) cannot be less than or equal to 0");
		double nD = (double) n;
		return ( (nD - 1) * currentMean + val ) / nD;
}
```

### PR DESCRIPTION
**Pull request by RohitJain1103**
Fixes #40 

**Changes**: Fixed the updateMean method in MathBasics class to run for all inputs with appropriate exception handling.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [X] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [X] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [X] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [X] My code does not contain any extra lines or extra spaces
- [X] I have requested reviews from other members

### CODE REVIEW

**Comment by RohitJain1103** [comment]
@CloudyPadmal @mariobehling @iamareebjamal Please review.

**Review by suresh-gandhi — APPROVED** [review]

File: app/src/main/java/io/neurolab/MathBasics.java [hunk 1]
```diff
@@ -20,14 +20,10 @@ public static double getMedian(double[] input) {
...
+            throw new IllegalArgumentException("Number of elements(n) cannot be less than or equal to 0");
         double nD = (double) n;
-        if (n > 1) {
-            double result = (double) ((1 / nD) * val);
-            result += (double) (((nD - 1) / (nD)) * currentMean);
-            return result;
-        }
-
-        return val;
+        return ( (nD - 1) * currentMean + val ) / nD;
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/MathBasics.java:26]: @RohitJain1103 I can see here that ` val ` was returned earlier. But here it is removed. Can you explain that?

**Review by jddeep — CHANGES_REQUESTED** [review]

**Review by RohitJain1103 — COMMENTED** [review]
.

File: app/src/main/java/io/neurolab/MathBasics.java (hunk 1 above)
Comment (RohitJain1103) [reply app/src/main/java/io/neurolab/MathBasics.java:26]: In the previous code val was returned as output when n <= 1 which is the same output in the new code if put n = 1. The formula on the last line is generalized formula.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**